api/jobs/handlers.py

```diff
@@ -482,6 +483,57 @@ def retry(self, _id):
         new_id = Queue.retry(j, force=True)
         return { "_id": new_id }
 
+
+    @require_drone
+    def prepare_complete(self, _id):
+        try:
+            j = Job.get(_id)
+        except Exception: # pylint: disable=broad-except
+            # TBD maybe raise APINotFound from Job.get?
+            raise APINotFoundException('Job not found')
+        payload = self.request.json
+        ticket = {
+            'job': j.id_,
+            'success': payload.get('success', False),
+        }
+        # TBD why not store success on the job itself?
+        return {'ticket': config.db.job_tickets.insert_one(ticket).inserted_id}
+
+
+    @require_login
+    def accept_failed_output(self, _id):
+        try:
+            j = Job.get(_id)
+        except Exception: # pylint: disable=broad-except
+            raise APINotFoundException('Job not found')
+
+        # Permission check
+        if not self.superuser_request:
+            j.destination.check_access(self.uid, 'rw')
+
+        if j.state != 'failed':
+            self.abort(400, 'Can only accept failed output of a job that failed')
+
+        # TBD recording user/timestamp? logging?
+        # TBD ticket cleanup/purging?
```

Comment:
We could always place a ttl index on the mongo collection.

Reply:
@ambrussimon and I discussed this earlier, I proposed 300 seconds.